server/src/internalClusterTest/java/org/elasticsearch/recovery/TruncatedRecoveryIT.java

```diff
@@ -102,20 +104,50 @@ public void testCancelRecoveryAndResume() throws Exception {
         indicesAdmin().prepareFlush().setForce(true).get(); // double flush to create safe commit in case of async durability
         indicesAdmin().prepareForceMerge().setMaxNumSegments(1).setFlush(true).get();
 
+        // We write some garbage into the shard directory so that we can verify that it is cleaned up before we resend.
+        // Cleanup helps prevent recovery from failing due to lack of space from garbage left over from a previous
+        // recovery that crashed during file transmission. #104473
+        // We can't look for the presence of the recovery temp files themselves because they are automatically
+        // cleaned up on clean shutdown by MultiFileWriter.
+        final String GARBAGE_PREFIX = "recovery.garbage.";
+
         final CountDownLatch latch = new CountDownLatch(1);
         final AtomicBoolean truncate = new AtomicBoolean(true);
+        IndicesService unluckyIndices = internalCluster().getInstance(IndicesService.class, unluckyNode);
         for (NodeStats dataNode : dataNodeStats) {
             MockTransportService.getInstance(dataNode.getNode().getName())
                 .addSendBehavior(
-                    internalCluster().getInstance(TransportService.class, unluckyNode.getNode().getName()),
+                    internalCluster().getInstance(TransportService.class, unluckyNode),
                     (connection, requestId, action, request, options) -> {
                         if (action.equals(PeerRecoveryTargetService.Actions.FILE_CHUNK)) {
                             RecoveryFileChunkRequest req = (RecoveryFileChunkRequest) request;
                             logger.info("file chunk [{}] lastChunk: {}", req, req.lastChunk());
                             if ((req.name().endsWith("cfs") || req.name().endsWith("fdt")) && req.lastChunk() && truncate.get()) {
```

Comment:
I'm wondering why we write a garbage file only under these conditions. Could we write the garbage file always, say on receipt of the first `FILE_CHUNK` message?

Reply:
We could. I think we just need a little extra bookkeeping so that we create garbage once per shard rather than once per test or once per chunk. Wasn't sure it was worth it - the filter here seemed like a reasonable proxy. But I can add the explicit bookkeeping if you prefer.

Reply:
Yeah I'd rather we were explicit here, otherwise future-me is going to wonder what's special about these particular `FILE_CHUNK` messages.